### PR TITLE
Parsing-sign-data-for-dApps

### PR DESCRIPTION
**Pull request by LendiHop**

### CODE REVIEW

File: src/hooks/parse-sign-payload/parse-sign-payload.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    let parsedPayload = '';
+    if (message.payload.match(TEZ_MSG_SIGN_PATTERN)) {
+      parsedPayload = emitMicheline(valueDecoder(Uint8ArrayConsumer.fromHexString(message.payload.slice(2))), {
+        indent: '  ',
+        newline: '\n'
+      }).slice(1, -1);
+    }
+
+    return parsedPayload;
+  };
```
Comment (unixvb) [inline src/hooks/parse-sign-payload/parse-sign-payload.ts:22]: create `parse-sign-payload.utils.ts` file and move this fn there

File: src/hooks/parse-sign-payload/taquito-local-forging.d.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+declare module '@taquito/local-forging/dist/lib/michelson/codec';
+declare module '@taquito/local-forging/dist/lib/uint8array-consumer';
```
Comment (unixvb) [inline src/hooks/parse-sign-payload/taquito-local-forging.d.ts:2]: why this?

File: src/hooks/parse-sign-payload/parse-sign-payload.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  useEffect(() => {
+    parseSignPayload().then(parsedPayload => setParsedPayload(parsedPayload));
```
Comment (unixvb) [inline src/hooks/parse-sign-payload/parse-sign-payload.ts:25]: short return

File: src/modals/confirmation-modal/d-app-operations-confirmation/sign-payload-request-confirmation/sign-payload-request-confirmation.tsx [hunk 4]
```diff
@@ -72,12 +81,16 @@ export const SignPayloadRequestConfirmation: FC<Props> = ({ message }) => {
...
         <View style={styles.descriptionContainer}>
-          <Divider size={formatSize(12)} />
           <Text style={styles.descriptionText}>Payload to sign</Text>
-          <Divider size={formatSize(12)} />
+          <TextSegmentControl
+            width={formatSize(138)}
+            selectedIndex={payloadTypeIndex}
+            values={['Raw', 'Bytes']}
+            onChange={setPayloadTypeIndex}
+          />
```
Comment (unixvb) [inline src/modals/confirmation-modal/d-app-operations-confirmation/sign-payload-request-confirmation/sign-payload-request-confirmation.tsx:90]: think about the case when parsing of payload message is going to be unsuccessful

**Review by unixvb — CHANGES_REQUESTED** [review]

File: src/hooks/parse-sign-payload/taquito-local-forging.d.ts (hunk 2 above)
Comment (LendiHop) [reply src/hooks/parse-sign-payload/taquito-local-forging.d.ts:2]: imports don't work without it

**Review by andrewkryshtal — APPROVED** [review]

File: src/utils/taquito-local-forging.utils.d.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+declare module '@taquito/local-forging/dist/lib/michelson/codec';
```
Comment (unixvb) [inline src/utils/taquito-local-forging.utils.d.ts:1]: rename this file to `parse-sign-payload.utils.d.ts` please

File: src/modals/confirmation-modal/d-app-operations-confirmation/sign-payload-request-confirmation/sign-payload-request-confirmation.tsx [hunk 6]
```diff
@@ -48,11 +51,19 @@ const approveSignPayloadRequest = (message: SignPayloadRequestOutput) =>
...
 
+const PAYLOAD_PREVIEW_TYPE_INDEX = 0;
+
 export const SignPayloadRequestConfirmation: FC<Props> = ({ message }) => {
   const styles = useSignPayloadRequestConfirmationStyles();
   const { goBack } = useNavigation();
   const accounts = useAccountsListSelector();
 
+  const payloadPreview = useParseSignPayloadHook(message);
+  const isPayloadParsed = isString(payloadPreview);
```
Comment (unixvb) [inline src/modals/confirmation-modal/d-app-operations-confirmation/sign-payload-request-confirmation/sign-payload-request-confirmation.tsx:62]: move this flag into `useParseSignPayloadHook`

**Review by unixvb — APPROVED** [review]

**Pull request merged**